packages/noya-renderer/src/components/SketchFileRenderer.tsx

```diff
@@ -193,16 +218,38 @@ export default memo(function SketchFileRenderer() {
       AffineTransform.identity,
     );
 
+    let selectedAndHighlightedLayers;
+    if (highlightedLayer.isMeasured) {
+      selectedAndHighlightedLayers = getSelectedAndHighlightedLayerBoundingRec();
+    }
+
     return (
       highlightedLayer && (
-        <HoverOutline
-          transform={layerTransform}
-          layer={layer}
-          paint={highlightPaint}
-        />
+        <>
+          <HoverOutline
+            transform={layerTransform}
+            layer={layer}
+            paint={highlightPaint}
+          />
+          {selectedAndHighlightedLayers &&
```

Comment:
Maybe it would be clearer if we rendered these elements separately from the `HoverOutline` in another `useMemo`?

In that case, we could potentially avoid a separate `useCallback` for `getSelectedAndHighlightedLayerBoundingRec`, and get the bounding rect / selected rect right before rendering these elements.

There might be some logic that we can extract out and share between both `useMemo`s, e.g. finding the layer transform. There might also be some logic that only applies to one or the other: e.g. the `highlightedLayer.precedence` only applies to the `HoverOutline`, but not the distance labels - so we shouldn't do an early return based on `highlightedLayer.precedence` when rendering `DistanceLabelAndPath`

Reply:
Removed element from `hoverOutline` and into its own `useMemo` callback `distanceLabelAndPathBetweenSketchLayers`. However I'm less familiar with this file, so please let me know if I can improve this change. Commit: https://github.com/noya-app/noya/pull/67/commits/83b58ff164b6619d64ef380f46ace1e60e73e4df